src/Raven.Client/Documents/LinqExtensions.cs

```diff
@@ -227,7 +227,7 @@ public static IRavenQueryable<TResult> ProjectInto<TResult>(this IQueryable quer
             var results = queryable.Provider.CreateQuery<TResult>(ofType.Expression);
             var ravenQueryInspector = (RavenQueryInspector<TResult>)results;
 
-            var membersList = ReflectionUtil.GetPropertiesAndFieldsFor<TResult>(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic).ToList();
+            var membersList = ReflectionUtil.GetPropertiesAndFieldsFor<TResult>(BindingFlags.Instance | BindingFlags.Public).ToList();
```

Comment:
I wonder if we shouldn't put `BindingFlags.Instance | BindingFlags.Public` as internal, read only, static value somewhere with meaningful name and use it on all places? This way we could have a single place where we control what types of properties / fields are checked when doing a query. What do you think?

Reply:
There's alot of places that you use in this enum and part of them are not the same as in here..
I think we should leave it like that.

Reply:
I was thinking about extracting it and use only for queries - basically the 3 places that you modified.

Reply:
Done